### ISSUE TITLE
handling revoked tokens

### ISSUE THREAD

**Issue by sull** [phediverse/mastodon-rest#8]
What is the approach to handle revoked access token? 

`401 Unauthorized`

How do I parse this response so I can redirect user to re-auth the app?

**Comment by iansltx** [comment]
Guzzle should throw a ClientException on promise resolution. If it's not doing that, that's more dev for me to do to figure out why and resolve. Probably should subclass or compose the exception into an AuthException of some sort anyway.

**Comment by sull** [comment]
Yep, I get the guzzle error response. Was looking for something more digestible. I am just using direct curl for now to get the simple json response.

**Comment by iansltx** [comment]
The Guzzle exception has a getResponse() method that you can use to pull the response:

https://github.com/guzzle/guzzle/blob/master/src/Exception/RequestException.php#L188

So you'd do `$e->getResponse()->getBody()->getContents()` or similar
